core/Http/RequestSanitizer.php

```diff
@@ -0,0 +1,233 @@
+                $value = $this->undoMagicQuotes($value);
+            }
+        } else if (!is_null($value)
+            && !is_bool($value)
+        ) {
+            throw new \Exception("The value to escape has not a supported type. Value = " . var_export($value, true));
+        }
+        return $value;
+    }
+
+    /**
+     * Unsanitizes a single value. Replaces escaped HTML entities with the actual characters.
+     *
+     * @param string $value
+     * @return string
+     */
+    public function unsanitizeValue($value)
+    {
+        return htmlspecialchars_decode($value, self::HTML_ENCODING_QUOTE_STYLE);
+    }
+
+    /**
+     * Sanitize an array of values recursively.
+     *
+     * @param mixed $value If an array, it is sanitized recursively. Keys are sanitized as well as values. If not,
+     *                     it is sanitized normally.
+     * @param bool $alreadyStripslashed Implementation detail, ignore.
+     * @return mixed
+     */
+    public function sanitizeValues($value, $alreadyStripslashed = false)
+    {
+        if (is_array($value)) {
+            foreach (array_keys($value) as $key) {
+                $newKey = $this->sanitizeValue($key, $alreadyStripslashed);
+
+                if ($key != $newKey) {
+                    $value[$newKey] = $value[$key];
+                    unset($value[$key]);
+                }
+
+                $value[$newKey] = $this->sanitizeValues($value[$newKey], $alreadyStripslashed);
+            }
+            return $value;
+        } else {
+            return $this->sanitizeValue($value);
+        }
+    }
+
+    /**
+     * Unsanitize array values recursively.
+     *
+     * @param mixed $values If an array, it is unsanitized recursively. Keys are not unsanitized. If not an array,
+     *                      it is sanitized normally.
+     * @return mixed
+     */
+    public function unsanitizeValues($value)
+    {
+        if (is_array($value)) {
+            $result = array();
+            foreach ($value as $key => $arrayValue) {
+                $result[$key] = $this->unsanitizeValues($arrayValue);
+            }
+            return $result;
+        } else {
+            return $this->unsanitizeValue($value);
+        }
+    }
+
+    /**
+     * @param string $value
+     * @return string Line breaks and line carriage removed
+     */
+    public function sanitizeLineBreaks($value)
+    {
+        return str_replace(array("\n", "\r"), '', $value);
+    }
+
+    /**
+     * Sanitizes a JSON string by value.
+     *
+     * @param string $value Parses a JSON string into an array and then sanitizes every element recursively. Array
+     *                      keys are sanitized as well.
+     * @return array
+     */
+    public function sanitizeJsonValues($value)
+    {
+        $value = $this->undoMagicQuotes($value);
+        $value = json_decode($value, $assoc = true);
+        return $this->sanitizeValues($value, $alreadyStripslashed = true);
```

Comment:
Can the value be put through `addslashes()` instead? That would avoid all the `alreadyStripslashed` parameters which kind of "pollute" the public API of the class with implementation detail. Not that it's a huge deal but if it can be easily avoided?